rust/hyperlane-base/src/types/multisig.rs

```diff
@@ -2,105 +2,54 @@ use std::collections::{hash_map::Entry, HashMap};
-    #[instrument(err, skip(self))]
-    pub async fn fetch_checkpoint(&self, index: u32) -> Result<Option<MultisigSignedCheckpoint>> {
-        // Keeps track of signed validator checkpoints for a particular root.
-        // In practice, it's likely that validators will all sign the same root for a
-        // particular index, but we'd like to be robust to this not being the case
-        let mut signed_checkpoints_per_root: HashMap<H256, Vec<SignedCheckpointWithSigner>> =
-            HashMap::new();
-
-        for (validator, checkpoint_syncer) in self.checkpoint_syncers.iter() {
-            // Gracefully ignore an error fetching the checkpoint from a validator's checkpoint syncer,
-            // which can happen if the validator has not signed the checkpoint at `index`.
-            if let Ok(Some(signed_checkpoint)) = checkpoint_syncer.fetch_checkpoint(index).await {
-                // If the signed checkpoint is for a different index, ignore it
-                if signed_checkpoint.checkpoint.index != index {
-                    continue;
-                }
-                // Ensure that the signature is actually by the validator
-                let signer = signed_checkpoint.recover()?;
-                if signer != *validator {
-                    continue;
-                }
-
-                // Insert the SignedCheckpointWithSigner into signed_checkpoints_per_root
-                let signed_checkpoint_with_signer = SignedCheckpointWithSigner {
-                    signer,
-                    signed_checkpoint,
-                };
-                let root = signed_checkpoint_with_signer
-                    .signed_checkpoint
-                    .checkpoint
-                    .root;
-
-                let signature_count = match signed_checkpoints_per_root.entry(root) {
-                    Entry::Occupied(mut entry) => {
-                        let vec = entry.get_mut();
-                        vec.push(signed_checkpoint_with_signer);
-                        vec.len()
-                    }
-                    Entry::Vacant(entry) => {
-                        entry.insert(vec![signed_checkpoint_with_signer]);
-                        1 // length of 1
-                    }
-                };
-                // If we've hit a quorum, create a MultisigSignedCheckpoint
-                if signature_count >= self.threshold {
-                    if let Some(signed_checkpoints) = signed_checkpoints_per_root.get(&root) {
-                        let checkpoint = MultisigSignedCheckpoint::try_from(signed_checkpoints)?;
-                        debug!(checkpoint=?checkpoint, "Fetched multisig checkpoint");
-                        return Ok(Some(checkpoint));
-                    }
-                }
-            }
-        }
-
-        Ok(None)
-    }
-
-    /// Attempts to get the latest index with a quorum of signatures among validators.
+    // TODO: Return an error if we can't find a checkpoint
```

Comment:
I assume related to https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/1486/files#r1057695383 ?